content/doc/administration/requirements/upgrade-java-guidelines.adoc

```diff
@@ -25,7 +25,7 @@ If you need to upgrade Jenkins as well as the JVM, we recommend that you:
 . Stop the Jenkins instance
 . Upgrade the JVM on which Jenkins is running
   ** Use a package manager to install the new JVM.
-  ** Make sure the default JVM is the newly installed version. If it is not, use the correct `java` command in the Jenkins startup scripts (`/etc/default/jenkins` or `/etc/init.d/jenkins`).
+  ** Make sure the default JVM is the newly installed version. If it is not, set `JAVA_HOME` or `JENKINS_JAVA_CMD` in the Jenkins startup scripts (`/lib/systemd/system/jenkins.service` and `/etc/systemd/system/jenkins.service.d/override.conf`).
```

Comment:
Do you need to modify the service unit file? I thought you were generally supposed to leave that alone and use the override file instead, but it’s been awhile since I’ve had to use systemd

Reply:
Yes you need to edit the override file using the information described in the main service unit. Feel free to suggest an improved wording.

Reply:
what about:
```suggestion
  ** Make sure the default JVM is the newly installed version. If it is not, set `JAVA_HOME` or `JENKINS_JAVA_CMD` in the systemd override file (`/etc/systemd/system/jenkins.service.d/override.conf`).
```